modules/ROOT/pages/partner-connector-upgrade.adoc

```diff
@@ -0,0 +1,406 @@
+
+MUnit 3.1 is compatible only with Mule runtime 4.3.0 and later. If your connector is compatible with Mule runtime 4.2.0 and earlier, you must create a legacy profile that overrides the MUnit version.
+
+Use the following bash script to test your custom connector against Java 17: 
+
+[source,bash]
+----
+#!/bin/bash
+RUNTIME_VERSION=4.6.0
+MUNIT_JVM=/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin/java
+mvn clean
+mkdir target
+mvn verify \
+   -DruntimeProduct=MULE_EE \
+   -DruntimeVersion=$RUNTIME_VERSION \
+   -Dmunit.jvm=$MUNIT_JVM \
+   -Dmule.module.tweaking.validation.skip=true \
+   -Dmule.jvm.version.extension.enforcement=LOOSE > ./target/test.log
+----
+
+=== Read Your Tests 
+
+After you run your MTF tests, your build has either of the following outcomes:
+
+* Test failures
++
+You will likely have to make changes to your custom connector code to ensure Java 17 compatibility.
+
+* All tests pass
++
+Either your custom connector does not require any major changes or your test suite is not comprehensive enough. You might want to review your test suite and double-check that your code coverage is good and that your test scenarios and assertions are not too simple.
+
+=== Communicate the Support Level of Your Custom Connector
+
+After you update your code and your tests are green, you are ready to release a new Java 17-compatible version of your custom connector. 
+
+. To communicate Java 17 compatibility, you must generate metadata for Java compatibility of your custom connector by adding or upgrading the custom connector’s mule-sdk-api dependency to the latest version: 
++
+[source,xml,linenums]
+----
+<dependency>
+   <groupId>org.mule.sdk</groupId>
+   <artifactId>mule-sdk-api</artifactId>
+   <version>0.7.0</version>
+</dependency>
+----
+
+. Add the `@JavaVersionSupport` annotation in the same class that has the `@Extension` annotation and include the JAVA_17 value, for example: 
++
+[source,java,linenums]
+----
+@Extension(name = "Database")
+@Operations(...)
+@JavaVersionSupport({JAVA_8, JAVA_11, JAVA_17})
+public class DatabaseConnector {
+..
+}
+----
+
+In Mule 4.5.0 and later, custom connectors that do not specify the `@JavaVersionSupport` annotation are assumed to be compatible with Java 8 and Java 11.
+
+You can mark your custom connector as compatible with Java 17 only, however, you must ensure there are no adoption or backward compatibility issues. 
+
+When you deploy a Mule app, Mule verifies that all modules in the Mule app are compatible with the Java version. If Mule finds an incompatibility, Mule throws an error and the application does not deploy. 
+
+If your code is compatible with Java 17 but you do not declare Java 17 compatibility, you can still get a successful test run. If you want to run a quick check on your custom connector or if one of your dependencies is not ready yet, you can pass an argument to avoid making hard checks on the Java support declaration.
```

Comment:
```suggestion
If your code is compatible with Java 17 but you do not declare Java 17 compatibility, you can still get a successful test run. To run a quick check on your custom connector or if one of your dependencies is not ready yet, you can pass an argument to avoid making hard checks on the Java support declaration.
```